WordPress/Classes/Categories/NSString+Helpers.m

```diff
@@ -126,9 +139,37 @@ - (NSString *)stringByReplacingHTMLEmoticonsWithEmoji
             [result replaceCharactersInRange:[match range] withString:replacement];
         }
     }
+
+    matches = [coreEmojiRegex matchesInString:result options:0 range:NSMakeRange(0, [result length])];
+    for (NSTextCheckingResult *match in [matches reverseObjectEnumerator]) {
+        NSRange range = [match rangeAtIndex:1];
+        NSString *filename = [result substringWithRange:range];
+        NSString *replacement = [self emojiUnicodeFromCoreEmojiFilename:filename];
+        if (replacement) {
+            [result replaceCharactersInRange:[match range] withString:replacement];
+        }
+    }
+
     return [NSString stringWithString:result];
 }
 
+/**
+ Pass the hex string for a unicode character and returns that unicode character. 
+ Core WordPress emoji images are named after their unicode code point. This makes
+ it convenient to scan the filename for its hex value and return the appropriate
+ emoji character.
+ */
+- (NSString *)emojiUnicodeFromCoreEmojiFilename:(NSString *)filename
```

Comment:
It's kind of weird that you'd pass a string to a string method. Shouldn't this be `- (NSString *)emojiUnicodeFromCoreEmojiFilename` and just use `self`?